### PR TITLE
Lambda lazy loading [TG-2485]

### PR DESCRIPTION
**Pull request by thk123**
Depends on: #1913 :white_check_mark: 

Adding some (failing) unit tests that demonstrate the problem with --lazy-methods when trying to get lambdas loaded.

### CODE REVIEW

File: unit/testing-utils/require_symbol.cpp
Comment (majakusber) [inline unit/testing-utils/require_symbol.cpp]: There is a `require_type::require_symbol` utility already. Consider using it here and given that your function checks for the existence too maybe the name of the function can reflect it? Something like `require_symbol_exists`.

File: unit/testing-utils/require_symbol.cpp
Comment (majakusber) [inline unit/testing-utils/require_symbol.cpp]: How about `lookup_ref`? The dereferenced value is returned anyway.

File: unit/java_bytecode/java_bytecode_parser/lambda_examples/ExternalLambdaAccessor.java
Comment (majakusber) [inline unit/java_bytecode/java_bytecode_parser/lambda_examples/ExternalLambdaAccessor.java]: Missing new line

**Review by majakusber — CHANGES_REQUESTED** [review]
Will continue with closer look at the unit tests tomorrow.

**Review by mgudemann — APPROVED** [review]
LGTM

File: unit/testing-utils/require_symbol.cpp
Comment (thk123) [reply unit/testing-utils/require_symbol.cpp]: I dereference after `REQUIRE`ing it is not null. If this `REQUIRE` fails it throws an exception that fails the test. The point of this function is to fail the test if the symbol is not found.

File: unit/testing-utils/require_symbol.cpp
Comment (thk123) [reply unit/testing-utils/require_symbol.cpp]: I don't have a `typet` just an identifier I want to look up so can't make use of that. However your name is much better so will use that.

**Comment by thk123** [comment]
@svorenova Comments adjusted apart from the `lookup_ref` which I don't think is appropriate for this utility.

**Review by majakusber — APPROVED** [review]
Thanks for the responses, I also looked at the unit tests closely, they look awesome :)

File: unit/testing-utils/free_form_cmdline.h
Comment (tautschnig) [inline unit/testing-utils/free_form_cmdline.h]: A space after `//` will make the linter happy

File: unit/testing-utils/require_symbol.h
Comment (tautschnig) [inline unit/testing-utils/require_symbol.h]: A space after `//` will make the linter happy.

**Review by tautschnig — COMMENTED** [review]
Just took a look at why Travis is failing, and put explanations in the comments below. Please fix before merging.

File: unit/testing-utils/free_form_cmdline.h
Comment (thk123) [reply unit/testing-utils/free_form_cmdline.h]: Fixed in the preceding PR so this will be fixed automatically when #1913 is merged and this is rebased ontop of develop

**Comment by thk123** [comment]
@tautschnig Thanks for the investigation - the issues you pointed out have actually been fixed already in the dependent PR #1913. I've added the label Do not merge to indicate that despite approval this is not actually ready to go int yet.

**Pull request merged**
